views/js/component/manager/manager.js

```diff
@@ -293,6 +298,16 @@ define([
                     e.stopPropagation();
                 });
 
+                //close the popup when clicking outside of the component
+                $(document).off('click.task-queue-manager').on('click.task-queue-manager', function(e){
+                    if($trigger.get(0) !== e.target && !$.contains($trigger.get(0), e.target)){
```

Comment:
When is this component rendered? If on page load it is a bit heavy to have this listener always on in `document`. Maybe you could set the listener in the `.show()` method and remove it in the `.hide()` method ?

Reply:
Good point !